### PR TITLE
swarm config file options take precedence over command line arguments #3444

### PR DESCRIPTION
**Pull request by jmozah**
for network Id, command line supplied value is given more precedence over the config.json value

### CODE REVIEW

**Comment by GitCop** [comment]
Thank you for your contribution! Your commits seem to not adhere to the repository coding standards

 * Commit: b30aab463831634a368e925a0fb87d6a828bf02d
  * Commits must be prefixed with the package(s) they modify

Please check the [contribution guidelines](https://github.com/ethereum/go-ethereum#contribution) for more details.

---
This message was auto-generated by https://gitcop.com

File: cmd/swarm/main.go [hunk 1]
```diff
@@ -252,6 +252,10 @@ func registerBzzService(ctx *cli.Context, stack *node.Node) {
 	if len(bzzport) > 0 {
 		bzzconfig.Port = bzzport
 	}
+	networkId := ctx.GlobalUint64(SwarmNetworkIdFlag.Name)
+	if networkId > 0 {
```
Comment (zelig) [inline cmd/swarm/main.go]: the problem is we assign the default value to this so its never gonna be 0
https://github.com/jmozah/go-ethereum/blob/b30aab463831634a368e925a0fb87d6a828bf02d/cmd/swarm/main.go#L79
so best to delete that line
and check in api/config that if network id is 0 then take default, `network.NetworkId` constant
same with other flags  like port possibly

**Comment by zelig** [comment]
@nolash,  @jmozah has already picked up this issue. Sorry for coordination fail ;)
please  see my comments above and here https://github.com/nolash/go-ethereum/commit/b520d7caa5f8a3307559db245974f0243b7908fd

**Comment by nolash** [comment]
@zelig no worries, for me useful anyhow to learn the code a bit!

**Comment by GitCop** [comment]
Thank you for your contribution! Your commits seem to not adhere to the repository coding standards

 * Commit: b30aab463831634a368e925a0fb87d6a828bf02d
  * Commits must be prefixed with the package(s) they modify

Please check the [contribution guidelines](https://github.com/ethereum/go-ethereum#contribution) for more details.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Thank you for your contribution! Your commits seem to not adhere to the repository coding standards

 * Commit: b30aab463831634a368e925a0fb87d6a828bf02d
  * Commits must be prefixed with the package(s) they modify

Please check the [contribution guidelines](https://github.com/ethereum/go-ethereum#contribution) for more details.

---
This message was auto-generated by https://gitcop.com

**Comment by zelig** [comment]
i guess i can close as of #3494

**Pull request closed without merging**